### PR TITLE
feat(G2G): add presence

### PR DESCRIPTION
**Pull request by Mattis44**
## Description 
Add a presence for G2G marketplace.
I added the discord presence support for the popular gamers' marketplace named G2G.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Presence changing from user choice.</summary>

image

</details>

<details>
<summary> Selected categories from user. </summary>

image

</details>

<details>
<summary> Viewing a profile. </summary>

image

</details>

<details>
<summary> View what user is buying. </summary>

image

</details>

### CODE REVIEW

**Comment by Mattis44** [comment]
weird fail, please check it

**Comment by Mattis44** [comment]
Last commit.

**Comment by Mattis44** [comment]
Last commit.

**Comment by Mattis44** [comment]
Last commit but weird fail. Please check it.

File: websites/G/G2G/presence.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+const presence = new Presence({
+		clientId: "1073167802296438875",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/XkzTOsp.png",
```
Comment (theusaf) [inline websites/G/G2G/presence.ts:8]: Should use the same 512x512 logo in the metadata
```suggestion
		largeImageKey: "https://i.imgur.com/U0FQhL3.png",
```

File: websites/G/G2G/presence.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		startTimestamp: browsingTimestamp,
+	};
+
+	switch (document.location.pathname.split("/")[1]) {
```
Comment (theusaf) [inline websites/G/G2G/presence.ts:12]: Properties of `document.location` are used many times. Consider destructuring them:
```ts
const { pathname, href, search } = document.location;
```

File: websites/G/G2G/presence.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+				{
+					label: "View Offer",
+					url: document.location.href,
+				},
+			];
+			break;
+		}
+		case "seller": {
+			presenceData.details = "Want to sell something";
+			presenceData.state = "Browsing...";
```
Comment (theusaf) [inline websites/G/G2G/presence.ts:46]: "Browsing..." does not make much sense here (and similar locations). Consider just having details and no state.
```suggestion
```

File: websites/G/G2G/presence.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		}
+		case "account": {
+			presenceData.details = "Viewing their account";
+			presenceData.state = "Browsing...";
+			break;
+		}
+		case "trending": {
+			presenceData.details = "Viewing a trending offer";
+			presenceData.state =
+				document.querySelector<HTMLBodyElement>("div.text-h4").textContent;
```
Comment (theusaf) [inline websites/G/G2G/presence.ts:62]: The wrong type is used in this query selector (and other ones)
```suggestion
				document.querySelector<HTMLDivElement>("div.text-h4").textContent;
```

File: websites/G/G2G/presence.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+			presenceData.buttons = [
+				{
+					label: "View Offer",
+					url: document.location.href,
+				},
+			];
+			break;
+		}
+		case "seller": {
+			presenceData.details = "Want to sell something";
```
Comment (theusaf) [inline websites/G/G2G/presence.ts:45]: The wording on this is a big strange.
Perhaps try something like:
* Selling something
* Looking to sell something
* Looking to sell

File: websites/G/G2G/metadata.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	"description": {
+		"en": "G2G is a world-leading digital marketplace, founded by serious technopreneurs."
+	},
+	"url": "www.g2g.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/U0FQhL3.png",
+	"thumbnail": "https://i.imgur.com/4JlsyFM.png",
+	"color": "#f03827",
+	"category": "games",
+	"tags": ["marketplace", "deals", "games"],
```
Comment (theusaf) [inline websites/G/G2G/metadata.json:17]: This file is not formatted correctly. Use `yarn format` to format it.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/G/G2G/metadata.json (hunk 6 above)
Comment (Mattis44) [reply websites/G/G2G/metadata.json:17]: Hello ! Thanks for your comments, i will take care of all you said. 
I already did a yarn format ^^

**Comment by Mattis44** [comment]
Last commit. 

Fixed all problems and tips reported above

(Again the fail presence task, check it.)

File: websites/G/G2G/metadata.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"id": "283757164923715587",
+		"name": "Mattis.#0001"
```
Comment (theusaf) [inline websites/G/G2G/metadata.json:5]: Discriminator added by outdated pmd
```suggestion
		"name": "Mattis."
```

**Review by theusaf — COMMENTED** [review]
Looks better. 
* The `metadata.json` file is not quite formatted correctly, (use `yarn format`, not just prettier) if possible.
* Make sure to mark completed conversations as resolved.

File: websites/G/G2G/metadata.json (hunk 7 above)
Comment (Mattis44) [reply websites/G/G2G/metadata.json:5]: Used yarn format on the terminal before, so i can't make it prettier.
And the discriminator is auto when you put your discord ID on the presence creating process

Comment (theusaf) [reply websites/G/G2G/metadata.json:5]: The discriminator is a bug from an old version of the presence creating program.

**Comment by Mattis44** [comment]
Final commit.

File: websites/G/G2G/presence.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+				{
+					label: "View Trending",
+					url: href,
+				},
+			];
+			break;
+		}
+		case "secure": {
+			presenceData.details = `Buying ${
+				document.querySelector<HTMLBodyElement>("div.col-8.row").textContent
```
Comment (theusaf) [inline websites/G/G2G/presence.ts:69]:
```suggestion
				document.querySelector<HTMLDivElement>("div.col-8.row").textContent
```

File: websites/G/G2G/metadata.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+		"id": "283757164923715587",
+		"name": "Mattis."
+	},
+	"service": "G2G",
+	"description": {
+		"en": "G2G is a world-leading digital marketplace, founded by serious technopreneurs."
+	},
+	"url": "www.g2g.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/XkzTOsp.png",
```
Comment (theusaf) [inline websites/G/G2G/metadata.json:13]: Use 512x512 logo
```suggestion
	"logo": "https://i.imgur.com/U0FQhL3.png",
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by Mattis44** [comment]
**Final commit.**

File: websites/G/G2G/metadata.json (hunk 9 above)
Comment (theusaf) [reply websites/G/G2G/metadata.json:13]: Doesn't look like these issues have been resolved.

Comment (Mattis44) [reply websites/G/G2G/metadata.json:13]: Yes, mistake. Sorry

**Comment by Mattis44** [comment]
Final commit.

File: websites/G/G2G/metadata.json (hunk 7 above)
Comment (EncryptedDev) [reply websites/G/G2G/metadata.json:5]: Unresolved

**Comment by Mattis44** [comment]
The discriminator is now resolved

File: websites/G/G2G/metadata.json [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+		"id": "283757164923715587",
+		"name": "Mattis."
+	},
+	"service": "G2G",
+	"description": {
+		"en": "G2G is a world-leading digital marketplace, founded by serious technopreneurs."
+	},
+	"url": "www.g2g.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/U0FQhL3.png",
```
Comment (Bas950) [inline websites/G/G2G/metadata.json:13]: Logo probs shouldn't have this white border 
image

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/G/G2G/metadata.json (hunk 10 above)
Comment (Mattis44) [reply websites/G/G2G/metadata.json:13]: took the logo of a reviewer. I will fix it.

Comment (Mattis44) [reply websites/G/G2G/metadata.json:13]: Ok so thats a github glitch. You can try download the logo.

Comment (Bas950) [reply websites/G/G2G/metadata.json:13]: Nvm, my bad

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**